omniduct/filesystems/s3.py

```diff
@@ -189,11 +191,73 @@ def _dir(self, path):
     #         if pattern is None or pattern.match(k.key[len(path_prefix):]):
     #             yield k.key
 
+    def _format_s3_path(self, path):
+        return 's3:' + path[4:] if path.startswith('s3n:') or path.startswith('s3a:') else path
+
+    def _is_s3(self, path):
+        """This function gives True if the given path is valid s3 path. This is used for the purpose of
+        validation."""
+        return 's3' in self._format_s3_path(path)
+
+    def _bucket_name(self, path):
```

Comment:
The bucket name is specified during the construction of the class, and is accessible via `self.bucket`